### ISSUE TITLE
coursier does not add `provided` libraries when scope is depended on explicitly  

### ISSUE THREAD

**Issue by RustedBones** [sbt/sbt#7626]
## steps

According to the [doc](https://www.scala-sbt.org/1.x/docs/Library-Management.html#Configurations), it should be possible to depend on the `provided` scope of a published library with

```sbt
libraryDependencies += "com.example" % "lib" % "version" % "compile->compile,provided"
```

This actually works when coursier is disabled with

```sbt
ThisBuild / useCoursier := false
```

See this https://github.com/spotify/magnolify/pull/1024/files

## problem

When using coursier, transitive `provided` libs are not added to the `compile` classpath, leading to a compile error.

## expectation

when setting ` "compile->compile,provided"` as configurations, `provided` libraries should be added to the classpath

**Comment by eed3si9n** [comment]
This looks like a duplicate of https://github.com/sbt/sbt/issues/4632 / https://github.com/coursier/coursier/issues/1846

**Comment by adpi2** [comment]
> This looks like a duplicate of https://github.com/sbt/sbt/issues/4632 

I don't think it is a duplicate of #4632. Here the user wants the `provided` dependencies to be downloaded, because the project explicitly depends on them. In #4632 the user wants some transitive dependencies to be ignored because they are marked as `provided` in some dependencies.


About https://github.com/coursier/coursier/issues/1846, I fixed it by adding a global Coursier option which is not yet supported by `sbt-coursier`. But notice that the option triggers a download of strictly all `provided` dependencies, direct and transitive ones. I am not sure if we need this in sbt. On this topic, I rather think sbt should behave as Maven does, and I need to dive in the Maven documentation to figure this out.
